### PR TITLE
let enqueue_job return job

### PR DESCRIPTION
**Pull request by stegben**
#87

### CODE REVIEW

File: arq/main.py [hunk 1]
```diff
@@ -120,18 +120,22 @@ def _bind_decorators(self):
...
             main_logger.debug('%s.%s → %s (called directly)', self.name, func_name, queue)
             data = self.job_class.encode(class_name=self.name, func_name=func_name, args=args, kwargs=kwargs)
             j = self.job_class(data, queue_name=queue)
             await getattr(self, j.func_name).direct(*j.args, **j.kwargs)
+        return j
 
-    def job_future(self, redis, queue: str, func_name: str, *args, **kwargs):
-        return redis.rpush(
+    async def job_future(self, redis, queue: str, func_name: str, *args, **kwargs):
+        data = self.job_class.encode(class_name=self.name, func_name=func_name, args=args, kwargs=kwargs)
```
Comment (samuelcolvin) [inline arq/main.py]: I don't like encoding then decoding here.

Better to modify `Job` to something like

```py
class Job:
    def __init__(self, *, job_id: str=None, queued_at: int=None, ...):
       self.id = self.generate_id(job_id)
       ...

    def encode(self):
        ...
    
    def decode(cls, raw_data: bytes, *, queue_name: str=None, raw_queue: bytes=None): -> Job:
        # like current __init__ method
```

File: tests/test_main.py [hunk 2]
```diff
@@ -203,7 +203,8 @@ class BadActor(Actor):
     class CustomActor(MockRedisDemoActor):
         queues = ['foobar']
     actor = CustomActor(loop=loop)
-    await actor.enqueue_job('add_numbers', 1, 1, queue='foobar')
+    job = await actor.enqueue_job('add_numbers', 1, 1, queue='foobar')
+    assert isinstance(job, Job)
```
Comment (samuelcolvin) [inline tests/test_main.py]: please add tests to check job id is either random or as set.

File: arq/main.py (hunk 1 above)
Comment (stegben) [reply arq/main.py]: I add this functionality in #92

**Comment by samuelcolvin** [comment]
replaced by #93

**Pull request closed without merging**